pycbc/inference/sampler_emcee.py

```diff
@@ -0,0 +1,995 @@
+        flatten : {True, bool}
+            The returned array will be one dimensional, with all desired
+            stats from all desired walkers concatenated together. If False,
+            the returned array will have dimension requested temps x requested 
+            walkers x requested iterations.
+        stats_group : {None, str}
+            The group in `fp` from which to retrieve the stats. If
+            None, searches in `fp.stats_group`.
+        array_class : {None, array class}
+            The type of array to return. The class must have a `from_kwargs`
+            class method. If None, will return a FieldArray.
+
+        Returns
+        -------
+        array_class
+            The likelihood stats, as an instance of the given
+            `array_class` (`FieldArray` if `array_class` is None).
+        """
+        if stats_group is None:
+            stats_group = fp.stats_group
+        if array_class is None:
+            array_class = FieldArray
+        fields = fp[stats_group].keys()
+        return cls._read_fields(fp, stats_group, fields, array_class,
+                thin_start=thin_start, thin_interval=thin_interval,
+                thin_end=thin_end, iteration=iteration, temps=temps,
+                walkers=walkers, flatten=flatten)
+
+    @classmethod
+    def compute_acls(cls, fp, start_index=None, end_index=None):
+        """Computes the autocorrleation length for all variable args for all
+        walkers for all temps in the given file. If the returned acl is inf,
+        will default to the number of requested iterations.
+
+        Parameters
+        -----------
+        fp : InferenceFile
+            An open file handler to read the samples from.
+        start_index : {None, int}
+            The start index to compute the acl from. If None, will try to use
+            the number of burn-in iterations in the file; otherwise, will start
+            at the first sample.
+        end_index : {None, int}
+            The end index to compute the acl to. If None, will go to the end
+            of the current iteration.
+
+        Returns
+        -------
+        WaveformArray
+            An ntemps x nwalkers `WaveformArray` containing the acl for each
+            walker and temp for each variable argument, with the variable
+            arguments as fields.
+        """
+        acls = {}
+        if end_index is None:
+            end_index = fp.niterations
+        tidx = numpy.arange(fp.ntemps)
+        widx = numpy.arange(fp.nwalkers)
+        for param in fp.variable_args:
+            these_acls = numpy.zeros((fp.ntemps, fp.nwalkers), dtype=int)
+            for tk in tidx:
+                for wi in widx:
+                    samples = cls.read_samples(fp, param,
+                            thin_start=start_index, thin_interval=1,
+                            thin_end=end_index,
+                            walkers=wi, temps=tk)[param]
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->